src/reverse_route.coffee

```diff
@@ -1,11 +1,22 @@
-class ReverseRoute extends String
+class ReverseRoute
 
   constructor: ({@path, @method}) ->
-    super @path
+    @length = @path.length
 
+    # Enable single character access
+    @[i] = character for character, i in @path
 
-  toString: ->
-    @path
+
+  # Forward String methods
+  indexOf: -> @path.indexOf arguments...
+
+  match: -> @path.match arguments...
+
+  slice: -> @path.slice arguments...
+
+  toString: -> @path
```

Comment:
That's what extending did, but V8 seems to do some black magic around native prototypes and the methods don't actually let you call them

Try this in chrome console:
```javascript
Foo = function() {}
Foo.prototype = String.prototype
foo = new Foo
foo.toString()  // Uncaught TypeError: String.prototype.toString is not generic
```